### PR TITLE
[Misc] Support DGL feature option.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

This PR should be a no-op PR.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,21 +10,106 @@ include(cmake/util/Util.cmake)
...
+
+  dgl_feature_option(
+      BUILD_TORCH
+      "Build the PyTorch plugin"
+      "none"
+  )
+  dgl_feature_option(
+      LIBCXX_ENABLE_PARALLEL_ALGORITHMS
+      "Enable the parallel algorithms library. This requires the PSTL to be available."
+      "none"
```
Comment (Rhett-Ying) [inline CMakeLists.txt:88]: "none" mean always `OFF` except user specifies via `-e "-Dxxx"`?

Comment (frozenbugs) [reply CMakeLists.txt:88]: Yes

File: CMakeLists.txt [hunk 2]
```diff
@@ -10,21 +10,106 @@ include(cmake/util/Util.cmake)
...
+  dgl_feature_option(
+      BUILD_GRAPHBOLT
+      "Build Graphbolt library"
+      "dev" "test"
+  )
+
+  dgl_feature_option(
+      BUILD_TORCH
+      "Build the PyTorch plugin"
+      "none"
```
[the remark this answers is not in the document]
Comment (frozenbugs) [reply CMakeLists.txt:83]: Why it is OFF in the CMakefile?

[the remark this answers is not in the document]
Comment (Rhett-Ying) [reply CMakeLists.txt:83]: it's probably due to tensoradaptor is optional in the past but for now, we always build it. @BarclayII could you help add more details?

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**